descartes_planner/src/planning_graph.cpp

```diff
@@ -998,15 +1005,20 @@ PlanningGraph::LinearWeightResult PlanningGraph::linearWeight(const JointTraject
       return result;
     }
 
-    double vector_diff = 0;
-    for (unsigned i = 0; i < start_vector.size(); i++)
+    if (custom_cost_function_)
     {
-      double joint_diff = std::abs(end_vector[i] - start_vector[i]);
-      vector_diff += joint_diff ;
+      result.second = custom_cost_function_(start_vector, end_vector);
+    }else{
```

Comment:
While you're at it, please put the braces on their own lines, e.g.

```
}
else
{
```

Just to keep with ROS style (I've been slowly trying to fix these issues in the older Descartes code).